### DISCUSSION TITLE
US State Map for WAS

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#1008] [Show and tell]
I decided to try to see how to implement a US State map within Cloudlog and have it working using the following JS library https://newsignature.github.io/us-map/

It only works for satellite QSOs /awards/was_map but the screenshot below shows how it works

![Screenshot from 2021-04-29 22-30-20](https://user-images.githubusercontent.com/84308/116621637-6655dd80-a93b-11eb-8744-adcd758f2c92.png)

- Green is confirmed
- Red is worked but not confirmed

I am looking for feedback on the colours should they actually be 

- Red for Unworked
- Yellow? for Worked but not confirmed
- Green for Confirmed?

I'm also wondering best way to display this @AndreasK79 what do you think? should there be a link with a popup it should be fairly easy to modify it to support HF etc too.

Feedback as ever welcome.

**Comment by magicbug** [comment]
![Screenshot from 2021-04-29 22-55-18](https://user-images.githubusercontent.com/84308/116623328-feed5d00-a93d-11eb-9564-00285f02b54c.png)

Made some colour changes after chatting with @m0pcb think this looks quite ok

**Comment by magicbug** [comment]
Very quickly snuck in a onclick button within WAS if you have selected "SAT" you should see "Show WAS Map"

Just a temp bodge

**Comment by magicbug** [comment]
![Screenshot from 2021-04-29 23-04-05](https://user-images.githubusercontent.com/84308/116628198-8343de00-a946-11eb-8845-2506c380c04a.png)

With deal data it looks quite good, this is my SAT WAS

**Comment by magicbug** [comment]
Made some further adjustments, you can now select the "band" after clicking "show".. click the "Show WAS Map"

![Screenshot from 2021-04-30 15-26-27](https://user-images.githubusercontent.com/84308/116710201-66042380-a9c9-11eb-8b50-1c39bd027f4e.png)

Nice to get this one crossed off the list.

**Comment by AndreasK79** [comment]
@magicbug glad you got it added. In my mind I had a thought of adding a tab to switch between table and map, much like the CQ award. A popup works too, but a nice thing would have been to click on a state and then show the QSOs in a popup. Also it would be nice to get a tooltip or something to tell which state it is for those that are not familiar with the location of every state. Other than that, great work!

**Reply by magicbug** [reply]
Yep, theres some event functions so doing popups won't be an issue, I did want to superimpose the state name on the map but can't seem todo that.

**Reply by AndreasK79** [reply]
By replacing the js-file with a fork from here https://github.com/kcaran/us-map, we can have these labels:
![image](https://user-images.githubusercontent.com/6977712/116806611-9a452480-ab2e-11eb-89d7-1e8f1aa869e1.png)

**Reply by AndreasK79** [reply]
And with a little tweaking, a tooltip:
![image](https://user-images.githubusercontent.com/6977712/116807257-7552b080-ab32-11eb-97f6-9aa44bf0ac87.png)

**Reply by magicbug** [reply]
@AndreasK79 on thats good didnt spot that fork

**Reply by AndreasK79** [reply]
Yep. But only the minified one worked.

**Reply by magicbug** [reply]
Minifed is OK thats what we should be using to keep page times fast

**Reply by magicbug** [reply]
I'll have a play with it this side see how it goes with my local copy

**Reply by AndreasK79** [reply]
@magicbug Please do. With a few lines of code, we can do this:
![image](https://user-images.githubusercontent.com/6977712/116818087-d3e75100-ab69-11eb-8f48-f77c006c92dc.png)
I can make a PR for labels and popup for QSO details. I have it running here now.

**Reply by magicbug** [reply]
Sounds good stick it as a PR!

**Reply by AndreasK79** [reply]
Done. I left the tooltip code there in case you want to play with it. It's disabled now, just need to remove 2 in the id in the div tag.

**Comment by M0TZO** [comment]
Looks good, but mine seems to be in a small vertical box in middle of screen. Tried with a different themes, but still remains a small box.
![image](https://user-images.githubusercontent.com/2846980/116828477-f2b00c80-ab96-11eb-8118-895ccaedcbcb.png)

**Reply by magicbug** [reply]
Clear your cache

**Reply by M0TZO** [reply]
Cheers. F12 & Empty Cache and Hard Reload worked.

**Comment by magicbug** [comment]
@AndreasK79 merged into the branch like the PR a lot, I've adjusted the red to make it more colourblind friendly but still think it needs adjusted more but better than it was.

Just need to make a nice legend for the colours next rather than the bullets.

**Comment by m0pcb** [comment]
I only see the "show map" button after I hit the "show" button, it's not
there on WAS page load.

May also be better to allow the WAS list to be resized or just not window
it and have the main page scroll... Just a thought...

73, Iain AD5XI / M0PCB

**Reply by AndreasK79** [reply]
Since we swapped the javascript-file, I would think you need to clear your cache in the browser.  Try CTRL-F5 in the browser.
